pynest/nest/lib/hl_api_connection_helpers.py

```diff
@@ -236,3 +248,61 @@ def fixdict(d):
     projections = fixdict(projections)
     sps(projections)
     sr('ConnectLayers')
+
+
+def _check_input_nodes(pre, post, conn_spec):
+    '''
+    Check the properties of `pre` and `post` nodes:
```

Comment:
This function does more than checking the properties, it also converts `pre` and `post` if needed. Perhaps it can be renamed to something like `_process_input_nodes`, as that would make it consistent with the helper functions for `syn_spec` and `conn_spec`.